core/src/main/scala/kafka/tools/MirrorMaker.scala

```diff
@@ -561,7 +561,12 @@ object MirrorMaker extends Logging with KafkaMetricsGroup {
       val consumerRebalanceListener = new InternalRebalanceListenerForNewConsumer(this, customRebalanceListener)
       if (whitelistOpt.isDefined) {
         try {
-          consumer.subscribe(Pattern.compile(whitelistOpt.get), consumerRebalanceListener)
+          val revisedWhitelist = whitelistOpt.get.trim
+            .replace(',', '|')
+            .replace(" ", "")
+            .replaceAll("""^["']+""","")
+            .replaceAll("""["']+$""","") // property files may bring quotes
+          consumer.subscribe(Pattern.compile(revisedWhitelist), consumerRebalanceListener)
```

Comment:
@vahidhashemian Agreed on your suggestion. And if we employs Whitelist here,  we need to additionally do a minor change for the exception capturing. Here is the analysis:

Before initializing Whitelist, the construction method for TopicFilter will be executed, which invokes Pattern.compile(regex). If an 'invalid' regex is offered, TopicFilter captures it and throws **RuntimeException**(not PatternSyntaxException), which means, later, the init method for MirrorMakerNewConsumer fails to capture this exception since it stares at PatternSyntaxException (although the impact is ignorable since the enclosing run method captures throwable)

Does it make sense? If you agree with the analysis above, I will make another commit following your suggestion and have this PR ready to be reviewed again. Is it okay to you?